### PR TITLE
Make the lint cmd output a bit easier to follow

### LINKED ISSUE #6247: Helm v3 - Linting multiple charts merges the error messages together
Output of `helm version`:
```
$ helm3 version
version.BuildInfo{Version:"v3.0.0-alpha.2", GitCommit:"97e7461e41455e58d89b4d7d192fed5352001d44", GitTreeState:"clean", GoVersion:"go1.12.7"}
```

Output of `kubectl version`:
N/A

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
N/A

When linting multiple charts in helm3 all error messages are merged together making it hard to work out which chart caused a failure.

```
$ mkdir -p /tmp/broken_chart_{1,2}
$ touch /tmp/broken_chart_1/Chart.yaml
$ helm3 lint /tmp/broken_chart*
Error: 1 chart(s) linted, 2 chart(s) failed
        %!s(<nil>)
        no chart found for linting (missing Chart.yaml)
        [ERROR] Chart.yaml: name is required
        [ERROR] Chart.yaml: directory name (broken_chart_1) and chart name () must be the same
        [ERROR] Chart.yaml: apiVersion is required. The value must be either "v1" or "v2"
        [ERROR] Chart.yaml: version is required
        [INFO] Chart.yaml: icon is recommended
        [INFO] values.yaml: file does not exist
        [WARNING] templates/: directory not found 
```

I'm not sure if this is the expected behavior of the lint action or not but this change makes it difficult to work out what failed compared to the same command in helm2.
```
$ helm lint /tmp/broken_chart*
==> Linting /tmp/broken_chart_1
[ERROR] Chart.yaml: name is required
[ERROR] Chart.yaml: directory name (broken_chart_1) and chart name () must be the same
[ERROR] Chart.yaml: version is required
[INFO] Chart.yaml: icon is recommended
[INFO] values.yaml: file does not exist
[WARNING] templates/: directory not found

==> Skipping /tmp/broken_chart_2
No chart found for linting (missing Chart.yaml)

Error: 1 chart(s) linted, 2 chart(s) failed
```

### PR DESCRIPTION
**Pull request by andytom**
**What this PR does / why we need it**:
Have tried to give the output of the lint command a bit of a clean up to
try to make it easier to follow. This splits the output by chart, moves
the summary to the end of the report rather than at the top and fixes
the number of failed charts count.

Closes #6247 

**Special notes for your reviewer**:

I have avoided making any changes to the lint action and tried to keep all the changes in the lint cmd, but happy to change this if there is a better suggestion.

Output before this
```
bin/helm lint /tmp/broken_chart_{1,2} /tmp/valid_chart/
Error: 2 chart(s) linted, 5 chart(s) failed
        name is required
        directory name (broken_chart_1) and chart name () must be the same
        apiVersion is required. The value must be either "v1" or "v2"
        version is required
        no chart found for linting (missing Chart.yaml)
        [ERROR] Chart.yaml: name is required
        [ERROR] Chart.yaml: directory name (broken_chart_1) and chart name () must be the same
        [ERROR] Chart.yaml: apiVersion is required. The value must be either "v1" or "v2"
        [ERROR] Chart.yaml: version is required
        [INFO] Chart.yaml: icon is recommended
        [INFO] values.yaml: file does not exist
        [WARNING] templates/: directory not found
        [ERROR] Chart.yaml: name is required
        [ERROR] Chart.yaml: directory name (broken_chart_1) and chart name () must be the same
        [ERROR] Chart.yaml: apiVersion is required. The value must be either "v1" or "v2"
        [ERROR] Chart.yaml: version is required
        [INFO] Chart.yaml: icon is recommended

```

Output after this
```
bin/helm lint /tmp/broken_chart_{1,2} /tmp/valid_chart/
==> Linting /tmp/broken_chart_1
[ERROR] Chart.yaml: name is required
[ERROR] Chart.yaml: directory name (broken_chart_1) and chart name () must be the same
[ERROR] Chart.yaml: apiVersion is required. The value must be either "v1" or "v2"
[ERROR] Chart.yaml: version is required
[INFO] Chart.yaml: icon is recommended
[INFO] values.yaml: file does not exist
[WARNING] templates/: directory not found
[ERROR] Chart.yaml: name is required
[ERROR] Chart.yaml: directory name (broken_chart_1) and chart name () must be the same
[ERROR] Chart.yaml: apiVersion is required. The value must be either "v1" or "v2"
[ERROR] Chart.yaml: version is required

==> Linting /tmp/broken_chart_2
Error no chart found for linting (missing Chart.yaml)

==> Linting /tmp/valid_chart/
[INFO] Chart.yaml: icon is recommended

Error: 3 chart(s) linted, 2 chart(s) failed
``` 
The duplicate `[ERROR] ...` rows will be fixed by #6296.

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: cmd/helm/lint.go [hunk 1]
```diff
@@ -56,20 +56,47 @@ func newLintCmd(out io.Writer) *cobra.Command {
...
+				// Adding extra new line here to break up the
+				// results, stops this from being a big wall of
+				// text and makes it easier to follow.
+				fmt.Fprint(&message, "\n")
 			}
+
 			fmt.Fprintf(out, message.String())
+
+			var summary strings.Builder
+			fmt.Fprintf(&summary, "%d chart(s) linted, %d chart(s) failed", len(paths), failed)
```
Comment (thomastaylor312) [inline cmd/helm/lint.go]: Non blocking nit: We probably don't need a string builder here, just print this string after the `if failed` block

**Review by thomastaylor312 — APPROVED** [review]
This is an awesome UX fix. Thanks for the PR!

File: cmd/helm/lint.go (hunk 1 above)
Comment (thomastaylor312) [reply cmd/helm/lint.go]: Because this is non-blocking, I merged. Should you wish to change it, feel free to open another PR!

Comment (andytom) [reply cmd/helm/lint.go]: Have opened #6373 to fix this.

**Pull request merged**